syntax = "proto3";
package common;

option go_package = "github.com/*company-data-covered*/services/go/pkg/generated/proto/common";

// RiskProtocolV1 represents the current primary causes in Risk Strat
// The names of the protocols have been changed slightly for readability
enum RiskProtocolV1 {
  RISK_PROTOCOL_V1_UNSPECIFIED = 0;
  RISK_PROTOCOL_V1_ABDOMINAL_PAIN = 1;
  RISK_PROTOCOL_V1_ALLERGIC_REACTION = 2;
  RISK_PROTOCOL_V1_CHEST_PAIN = 3;
  RISK_PROTOCOL_V1_CONFUSION = 4;
  RISK_PROTOCOL_V1_DIZZINESS = 5;
  RISK_PROTOCOL_V1_EPISTAXIS = 6;
  RISK_PROTOCOL_V1_FEVER = 7;
  RISK_PROTOCOL_V1_HIGH_RISK_UNDER_18 = 8;
  RISK_PROTOCOL_V1_HIGH_RISK_OVER_18 = 9;
  RISK_PROTOCOL_V1_LETHARGY = 10;
  RISK_PROTOCOL_V1_HYPOTENSION = 11;
  RISK_PROTOCOL_V1_NEUROLOGIC_COMPLAINT = 12;
  RISK_PROTOCOL_V1_NUMBNESS = 13;
  RISK_PROTOCOL_V1_OVERDOSE_POISONING = 14;
  RISK_PROTOCOL_V1_PACEMAKER_COMPLICATIONS = 15;
  RISK_PROTOCOL_V1_SELF_HARM = 16;
  RISK_PROTOCOL_V1_SHORTNESS_OF_BREATH = 17;
  RISK_PROTOCOL_V1_SYNCOPE = 18;
  RISK_PROTOCOL_V1_TESTICULAR_PAIN = 19;
  RISK_PROTOCOL_V1_UNABLE_TO_URINATE = 20;
  RISK_PROTOCOL_V1_WEAKNESS = 21;
  RISK_PROTOCOL_V1_DIFFICULTY_URINATING = 22;
  RISK_PROTOCOL_V1_EXTREMITY_SWELLING = 23;
  RISK_PROTOCOL_V1_FALL_WITHOUT_INJURY = 24;
  RISK_PROTOCOL_V1_HEAD_INJURY = 25;
  RISK_PROTOCOL_V1_HEADACHE = 26;
  RISK_PROTOCOL_V1_PALPITATIONS_ABNORMAL_HEART_RATE = 27;
  RISK_PROTOCOL_V1_ANXIETY = 28;
  RISK_PROTOCOL_V1_BLADDER_CATHETER_ISSUE = 29;
  RISK_PROTOCOL_V1_BLOOD_IN_STOOL = 30;
  RISK_PROTOCOL_V1_BLOOD_SUGAR_CONCERNS = 31;
  RISK_PROTOCOL_V1_DEHYDRATION = 32;
  RISK_PROTOCOL_V1_DIARRHEA = 33;
  RISK_PROTOCOL_V1_EXTREMITY_INJURY_OR_PAIN = 34;
  RISK_PROTOCOL_V1_NAUSEA_OR_VOMITING = 35;
  RISK_PROTOCOL_V1_RECTAL_PAIN = 36;
  RISK_PROTOCOL_V1_SKIN_RASH = 37;
  RISK_PROTOCOL_V1_ANIMAL_BITE = 38;
  RISK_PROTOCOL_V1_BACK_PAIN = 39;
  RISK_PROTOCOL_V1_BURN = 40;
  RISK_PROTOCOL_V1_CONSTIPATION = 41;
  RISK_PROTOCOL_V1_COUGH = 42;
  RISK_PROTOCOL_V1_DENTAL_PAIN = 43;
  RISK_PROTOCOL_V1_EAR_PAIN = 44;
  RISK_PROTOCOL_V1_GASTRIC_TUBE_ABNORMALITY = 45;
  RISK_PROTOCOL_V1_GENERAL_COMPLAINT = 46;
  RISK_PROTOCOL_V1_HYPERTENSION = 47;
  RISK_PROTOCOL_V1_LABORATORY_TEST_CONCERN = 48;
  RISK_PROTOCOL_V1_LACERATION = 49;
  RISK_PROTOCOL_V1_POST_ACUTE_PATIENT_FOLLOW_UP = 50;
  RISK_PROTOCOL_V1_SINUS_PAIN = 51;
  RISK_PROTOCOL_V1_SORE_THROAT = 52;
  RISK_PROTOCOL_V1_WOUND_EVALUATION = 53;
  RISK_PROTOCOL_V1_VISION_PROBLEM = 54;
  // The following protocols were generated by looking up the most populous
  // risk strat names over the last 2 months (from ~Jun-Aug 10th 2022) Getting
  // clinical to provide an estimate of what their acuity is and list it in a
  // spreasheet:
  // https://*company-data-covered*-my.sharepoint.com/:x:/p/enrique_hernandez/EZNLvAyh6fFNqtKlpJfRhfIBrVTdINK5Mk4L5OeDIsgY-A?e=z8cmI2
  RISK_PROTOCOL_V1_FLU_LIKE_SYMPTOMS = 55;
  RISK_PROTOCOL_V1_EDUCATIONAL_PROGRAM = 56;
  RISK_PROTOCOL_V1_SEIZURE = 57;
  RISK_PROTOCOL_V1_HOSPICE = 58;
  RISK_PROTOCOL_V1_PULSE_HEART = 59;
  RISK_PROTOCOL_V1_RESPIRATORY_COMPLAINT_SENIOR_LIVING = 60;
  RISK_PROTOCOL_V1_ADVANCED_CARE_MONOCLONAL_ANTIBODIES = 61;
  RISK_PROTOCOL_V1_ASYMPTOMATIC_COVID_TESTING = 62;
  RISK_PROTOCOL_V1_HALLUCINATIONS = 63;
  RISK_PROTOCOL_V1_GYNECOLOGICAL = 64;
  RISK_PROTOCOL_V1_PROVIDER_REQUESTED = 65;
  RISK_PROTOCOL_V1_EMS_ET3_TREAT_IN_PLACE = 66;
  RISK_PROTOCOL_V1_ADVANCED_CARE = 67;
  RISK_PROTOCOL_V1_NECK_PAIN = 68;
  RISK_PROTOCOL_V1_EMERGENCY_DEPARTMENT_TO_HOME = 69;
  RISK_PROTOCOL_V1_REGISTERED_NURSE = 70;
  // The following protocols are from a clinical effort to consolidate similar
  // protocols, requiring updated acuity logic. Notes from the conversation were
  // included in the following pull request:
  // https://github.com/*company-data-covered*/services/pull/2118
  RISK_PROTOCOL_V1_NAUSEA_OR_VOMITING_OR_DIARRHEA = 71;
  RISK_PROTOCOL_V1_ABDOMINAL_PAIN_OR_CONSTIPATION = 72;
  RISK_PROTOCOL_V1_COUGH_OR_URI_OR_SINUSITIS_OR_SORE_THROAT = 73;
  RISK_PROTOCOL_V1_NECK_BACK_SPINE_PAIN = 74;
  RISK_PROTOCOL_V1_PREGNANCY_GYNECOLOGICAL = 75;
  RISK_PROTOCOL_V1_ANXIETY_PANIC_HALLUCINATION = 76;
  RISK_PROTOCOL_V1_PALPITATIONS_ABNORMAL_HEART_RATE_PACEMAKER = 77;
  RISK_PROTOCOL_V1_WEAKNESS_DEHYDRATION = 78;
  RISK_PROTOCOL_V1_NEUROLOGIC_NUMBNESS = 79;
  // The following protocols were added as part of the Acuity Model V2 changes
  // for validation and experimentation.
  // One pager:
  // https://*company-data-covered*.sharepoint.com/:w:/s/tech-team/Eci1FUFPrQBNiv8q1QPLN8gBjrhn1g1yzrR9HEwpA8915w?e=noYeAg
  // Proposed changes:
  // https://*company-data-covered*.sharepoint.com/:x:/s/tech-team/EWw0ysdf2LpImaP-TvHOSKkBVpKx_5mwdzkzNk6LyX_uXg?e=Z5nCLg
  // Experiment:
  // https://*company-data-covered*.sharepoint.com/:w:/s/tech-team/EeBHPWAa7iZCkGKQV0lvw40Bgi69fwE-eUh-AKeOqhqdvg?e=lvTR92
  RISK_PROTOCOL_V1_DH_PROVIDER_RISK_STRATIFICATION = 80;
  RISK_PROTOCOL_V1_COVID19_FACILITY_TESTING = 81;
  RISK_PROTOCOL_V1_OPTUM_SKILLED_NURSING_FACILITIES_ONBOARDING_PROCESS = 82;
  RISK_PROTOCOL_V1_PARTNER_USING_NATIONALLY_ACCREDITED_RISK_STRAT_PROCESS = 83;
  RISK_PROTOCOL_V1_TOUCHLESS_WEB_ONBOARDING = 84;
  RISK_PROTOCOL_V1_WELLNESS_VISIT_HEDIS_PROACTIVE_APPOINTMENT = 85;
}
